### PR TITLE
Fix #40, #91, #92 - Migrate CI to GitHub Actions

### LINKED ISSUE #40: Drop EOL'd Rails versions
As Rails 6 is shipped(#39) - we want to drop support for Rails 4.2 and ruby versions older than 2.2.2.

### PR DESCRIPTION
**Pull request by petergoldstein**
This PR migrates to GitHub Actions for CI, drops support for older Rubies and Rails versions, and adds more recent versions to CI.

Runs green on my fork.

### CODE REVIEW

**Comment by teoljungberg** [comment]
Actions are enabled. I don’t think there’s anything else I need to do?

I’d appreciate if you focused this PR solely on replacing Travis with GitHub actions, we’ll do follow ups for dropping old Rails and Ruby versions.

**Review by teoljungberg — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            gemfile: gemfiles/rails60.gemfile
+          - ruby: head
+            gemfile: gemfiles/rails52.gemfile
+          - ruby: head
+            gemfile: gemfiles/rails60.gemfile
+        os:
+          - ubuntu-latest
+    services:
+      postgres:
+        image: postgres
```
[the remark this answers is not in the document]
Comment (petergoldstein) [reply .github/workflows/ci.yml]: Done

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      - uses: actions/checkout@v3
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Setup environment
+        run: bin/setup
+      - name: Run tests
+        run: bundle exec rake
```
[the remark this answers is not in the document]
Comment (petergoldstein) [reply .github/workflows/ci.yml]: Done

**Comment by teoljungberg** [comment]
I’d want this addressed too

https://github.com/teoljungberg/fx/pull/93#issuecomment-1156763338

**Comment by petergoldstein** [comment]
@teoljungberg Not doing these changes all at once makes this harder.  Essentially I would need to restructure the CI matrix multiple times. dealing with EOLd versions of Ruby and Rails and ensuring they work in the GitHub Actions environment.  And then, after that's merged, making the changes to drop / add those Ruby and Rails versions to get to ... the version of the CI matrix in this PR as it stands.  Dig a hole, fill it back in.

Given that CI is non-functional on this repo right now, in my view that approach is not sensible.  This PR is the target state.

I'm happy to make other adjustments (lock Postgres to 14, fix a trailing newline), but I'm not going to segment out the Ruby and Rails version changes from this PR.  That said, feel free to take this PR and do what you want - merge it, close it, or use it as the basis for the set of changes described above.

**Comment by teoljungberg** [comment]
I understand that, thank you for letting me know. I opened this as 3 separate issues as I wanted them to be just that, that way we can merge the removal of old ruby versions, wait and see if anyone complains, release a new version, let that sit for a few weeks to let people’s dependabots get the new release in.

Then do the same cycle for adjusting the Rails support with dropping old versions, and introducing the missing new ones.

No hard feelings if you don’t want to do that.

Thanks for your work in this, I’ll move this PR back into a draft and go through this when I find the time.

**Comment by petergoldstein** [comment]
Ok.  That's fine.

As a note, the proposed approach doesn't actually accomplish anything.  This PR is a CI only change, so it doesn't break any Ruby or Rails versions that are currently working.  Merging it - or not merging - will not cause any users to complain, because from their perspective there will be no difference.

If you wanted to accomplish something like what you propose, you would:

1. Bump the minimum version of `activerecord` and `railties` in the gemspec from the current version of `>= 4.0.0` to `>= 5.2.0`
2. Release a new major version with that change
3. Bump the minimum version of Ruby listed in the gemspec from the current value of `>=2.1` to `>=2.7`
4. Release a new major version with that change.

Note that you'll need to bump `activerecord` and `railties` first, as there is no Rails version less than `5.2.0` that runs with a Ruby version of `>= 2.7` -  https://www.fastruby.io/blog/ruby/rails/versions/compatibility-table.html .

Alternately, you could combine steps 1 and 3, and simply release a single new major version. 

This PR can be merged at any time, and it doesn't impact the end user - it just impacts what's actually tested.

**Comment by petergoldstein** [comment]
Since this is still open I've updated it so it includes:

- Running Rails head and Ruby 2.7 in combination, since that's not supported
- Ruby 3.2, which was released in Dec 2022.

All my comments above still apply.

**Comment by teoljungberg** [comment]
All right, I decided to roll with this. This project needs a CI, and this works well so I'll set this up!

I'll merge this without the dependabot upgrade, as that is needed less for this project.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+          rubygems: latest
+      - name: Setup environment
+        run: bin/setup
+      - name: Run tests
+        run: bundle exec rake
+
```
Comment (teoljungberg) [inline .github/workflows/ci.yml:72]:
```suggestion
```

**Review by teoljungberg — APPROVED** [review]

**Pull request merged**